Feature: Allow for a single object upload for posts

duplixx opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.
-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

Fixes #1127 

**Did you add tests for your changes?**

<!--Yes or No. Note: Add unit tests or automation tests for your code.-->

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/e8c4ecdb-26eb-4d40-a834-d0de0c248f47




**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->

duplixx commented:
@aashimawadhwa, can you review it and let me know if any changes are required?

aashimawadhwa commented:
can you fix the linting workflow? @duplixx

palisadoes commented:
The linting config file has been reverted. Please merge with your upstream and make an updating commit.

aashimawadhwa commented:
@duplixx  can you fix the failing test.

duplixx commented:
I'm currently stuck in an error that doesn't make sense. Could you please take a look at it, @aashimawadhwa, @palisadoes, @noman2002? I'll be attaching a video for better understanding.
https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/8cac51cd-3faa-4234-b027-a18b5d326d5f

duplixx commented:
@aashimawadhwa @palisadoes Great news! I fixed the error by running it on a different machine. Other than that, we may need to add more test cases in the future. For now, I've included the test cases for the code I added.

duplixx commented:
hi @aashimawadhwa @EshaanAgg @noman2002 can you review this 
thanks!!

Cioppolo14 commented:
@duplixx  There are still failing tests.  Are you working on this or stuck?

aashimawadhwa commented:
@duplixx  updates?

duplixx commented:
Hi @aashimawadhwa, I am working in it to resolve the issue, but any help will be greatly appreciated.

rishav-jha-mech commented:
@duplixx while running tests 

```sh
./coverage/lcov-report/index.html
```

run this file on live server, so that you can see which block of code is not being executed/covered by the tests. 

Also you while running tests you can use this command, 
```sh
npm run jest preview
```

to see how the UI changes, or what effects take place for a particular test. This is the standard way to write tests and debug in talawa-admin.

rishav-jha-mech commented on public/locales/en.json:
```diff
@@ -288,6 +288,7 @@
     "postDetails": "Post Details",
     "postTitle1": "Write title of the post",
     "postTitle": "Title",
+    "addMedia": "Upload Photo or Video",
```
If you are adding any new text in the project, please make sure to add it for other languages too.

duplixx replied:
Thanks @rishav-jha-mech, i'll add it in different languages🙌

rishav-jha-mech commented on src/screens/OrgPost/OrgPost.tsx:
```diff
@@ -110,10 +114,10 @@ function orgPost(): JSX.Element {
-      /* istanbul ignore next */
+
       if (data) {
         toast.success('Congratulations! You have Posted Something.');
```
You can add a multilingual message here.

duplixx commented:
Hey @rishav-jha-mech, I have added multilingual support for text rendering. I'm facing difficulty in implementing test cases for toasters. If you have any hints on how to add test cases for that, it would be helpful.

rishav-jha-mech commented:
@duplixx you need to create a mock fn for Toast, go to this file and see how it is implemented, you will get the idea.
 `src\components\OrgPostCard\OrgPostCard.test.tsx`

duplixx commented:
![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/5d29c058-ae13-49e3-a5fc-3b39ed5a5560)
![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/43d84c17-363d-43f2-b262-36dcbd44fcfa)

![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/7b2155cf-d272-44df-98ca-dae9ebd3abc1)


Hi @rishav-jha-mech 
I was adding a test case for this line, and the test case I added looks fine, but when I tried running it, it showed an error.

![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/8bab54f1-b029-4dcb-a404-0649d2053f06)

duplixx commented:
Hi @rishav-jha-mech,
I've implemented the functionality and made the necessary changes. We can open an issue later to ensure complete test coverage of 100%.

rishav-jha-mech commented:
@duplixx why do you wanna merge into main ?, set the base branch to develop and fix the tests

duplixx commented:
@rishav-jha-mech should i open pr in another branch to fix this ??

rishav-jha-mech commented:
@duplixx you can open a new PR for this, keep this PR as a reference to the new one.

noman2002 commented:
@duplixx Should this be closed ??

duplixx commented:
yeah sure @noman2002
